File: ruby/abc192_c.rb
n, k = gets.split(" ").map(&:to_i)
(puts n; exit) if k==0
a = n.to_s.split("").map(&:to_i)
f = 0
k.times do
  g1 = a.sort.reverse.join.to_i
  g2 = a.sort.join.to_i
  f = g1-g2
  a = f.to_s.split("").map(&:to_i)
end
puts f

File: csharp/abc192_c.cs
using System;
using System.Linq;
class ABC
{
  static void Main()
  {

    string[] s = Console.ReadLine().Split(" ");
    int n = int.Parse(s[0]);
    int k = int.Parse(s[1]);
    while(k > 0){
      char[] t = n.ToString().ToCharArray();
      Array.Sort(t);
      int g2 = int.Parse(String.Join("", t));
      Array.Reverse(t);
      int g1 = int.Parse(String.Join("", t));
      n = g1 - g2;
      k--;
    }
    Console.WriteLine(n);
  }
}